frontend/packages/client/src/hooks/useJoinCommunity.js

```diff
@@ -1,27 +1,21 @@
-import { useReducer, useCallback } from 'react';
-import { defaultReducer, INITIAL_STATE } from '../reducers';
+import { useMutation, useQueryClient } from 'react-query';
 import { useErrorHandlerContext } from '../contexts/ErrorHandler';
 import { getCompositeSigs } from 'utils';
 
 export default function useJoinCommunity() {
-  const [state, dispatch] = useReducer(defaultReducer, {
-    ...INITIAL_STATE,
-    loading: false,
-  });
+  const queryClient = useQueryClient();
   const { notifyError } = useErrorHandlerContext();
 
-  const createCommunityUser = useCallback(
-    async (communityId, user, injectedProvider) => {
+  const { mutateAsync: createCommunityUserMutation } = useMutation(
```

Comment:
I had tu use `mutateAsync` as opposed to `mutate` because of the way the hook is used in the component. The component is expecting to get a result from the mutation like here: https://github.com/DapperCollectives/CAST/pull/108/files#diff-ba7b9f47b489056dd6b487fee2e4bf14d1d836b8f8de40956a83cd57e1e84b39R34

In the docs: https://react-query.tanstack.com/reference/useMutation  retuning the mutate function seems the way to go, they we could use something like:
```
const { mutate, data, error, isLoading } = useSomeWrapperFunctionMutation(mutate, 
{
   onSuccess: () => {
        manage side effects here; general ones like some query invalidation adding data to loaded queries and so on
   }
});
   
then we can call
mutate(payload);

or if we have specific side effects:
mutate(payload, {
   onSuccess: (data, variables, context) => {
     // something we need to happen on the component that triggered the update
   },
   onError: (error, variables, context) => {
     // same here
   },
   onSettled: (data, error, variables, context) => {
     // same here
   },
 })